libminifi/test/unit/JsonFlowSerializerTests.cpp

```diff
@@ -0,0 +1,425 @@
+                    "CA Certificate": "certs/ca-cert.pem",
+                    "Use System Cert Store": "false"
+                },
+                "propertyDescriptors": {
+                    "Private Key": {
+                        "name": "Private Key",
+                        "identifiesControllerService": false,
+                        "sensitive": false
+                    },
+                    "Client Certificate": {
+                        "name": "Client Certificate",
+                        "identifiesControllerService": false,
+                        "sensitive": false
+                    },
+                    "Passphrase": {
+                        "name": "Passphrase",
+                        "identifiesControllerService": false,
+                        "sensitive": true
+                    },
+                    "CA Certificate": {
+                        "name": "CA Certificate",
+                        "identifiesControllerService": false,
+                        "sensitive": false
+                    },
+                    "Use System Cert Store": {
+                        "name": "Use System Cert Store",
+                        "identifiesControllerService": false,
+                        "sensitive": false
+                    }
+                },
+                "controllerServiceApis": [
+                    {
+                        "type": "org.apache.nifi.minifi.controllers.SSLContextService",
+                        "bundle": {
+                            "group": "org.apache.nifi.minifi",
+                            "artifact": "minifi-system",
+                            "version": "1.23.06"
+                        }
+                    }
+                ],
+                "componentType": "CONTROLLER_SERVICE",
+                "groupIdentifier": "8b4d66dc-9085-4722-b35b-3492f363baa3"
+            }
+        ],
+        "variables": {},
+        "componentType": "PROCESS_GROUP"
+    }
+}
+)";
+
+const utils::crypto::Bytes secret_key = utils::string::from_hex("75536923a75928a970077f9dae2c2b166a5413e020cb5190de4fb8edce1a38c7");
+const utils::crypto::EncryptionProvider encryption_provider{secret_key};
+
+TEST_CASE("JsonFlowSerializer can encrypt the sensitive properties") {
+  ConfigurationTestController test_controller;
+  core::flow::AdaptiveConfiguration json_configuration{test_controller.getContext()};
+  const auto process_group = json_configuration.getRootFromPayload(std::string{config_json});
+  REQUIRE(process_group);
+
+  const auto schema = core::flow::FlowSchema::getNiFiFlowJson();
+
+  rapidjson::Document doc;
+  rapidjson::ParseResult res = doc.Parse(config_json.data(), config_json.size());
+  REQUIRE(res);
+  const auto flow_serializer = core::json::JsonFlowSerializer{std::move(doc)};
+
+  std::string config_json_encrypted = flow_serializer.serialize(*process_group, schema, encryption_provider);
+
+  SECTION("check the encryption of invokehttp-proxy-password") {
```

Comment:
Do we need separate sections for each encrypted property? I think it would be simpler to check them both in a single run, extracting the checker function into a lambda function and run it with different regex and expected decrypted value value parameters. This way we don't run the serialization twice. Same could be done for the yaml tests.

Reply:
I created the sections in order to give names to the code blocks, but you're right that they are not necessary, and make the tests slower.  I have removed them in 9613542f2ccb5008be30717d0da4d799c45b7956.